web/src/components/PasswordWithMeter.js

```diff
@@ -0,0 +1,105 @@
+import { Choice, TextField } from '@cmsgov/design-system-core';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import zxcvbn from 'zxcvbn';
+
+class Password extends Component {
+  state = { showPassword: false, strength: 0 };
+
+  componentWillMount() {
+    const { value } = this.props;
+    if (value) {
+      this.changePassword({ target: { value } });
+    }
+  }
+
+  changePassword = e => {
+    const { compareTo, onChange } = this.props;
+    this.setState({ strength: zxcvbn(e.target.value, compareTo).score });
+
+    if (onChange) {
+      onChange(e);
+    }
+  };
+
+  toggleShowPassword = () => {
+    this.setState(prev => ({ showPassword: !prev.showPassword }));
+  };
+
+  render() {
+    const { title, value, ...rest } = this.props;
+    const { showPassword, strength } = this.state;
+
+    let passwordQuality = 'Password strength: Weak';
+    if (value.length === 0) {
+      passwordQuality = <span>&nbsp;</span>;
+    } else if (strength === 3) {
+      passwordQuality = 'Password strength: Good';
+    } else if (strength > 3) {
+      passwordQuality = 'Password strength: Great';
+    }
+
+    const strengthClass = [
+      'strength-meter',
+      value.length > 0 ? 'visible' : 'hidden'
+    ]
+      .join(' ')
+      .trim();
+
+    const qualityClass = [
+      'strength-meter-quality',
+      value.length > 0 ? 'visible' : 'hidden'
+    ]
+      .join(' ')
+      .trim();
+
+    return (
+      <div {...rest}>
+        <div className="password-with-meter">
+          <Choice
```

Comment:
I'm a little concerned at this coming before the associated `TextField` in the source. This means that screen readers are going to read the "Show Password?" checkbox prior to announcing that the user is about to enter a password. Maybe we don't use the `TextField` for this one so we can put things in a more sensible order?

Reply:
Ah, great point.  I hadn't even thought about that.  Can we plan on pairing on this one too?  I'm not sure what'd make a proper fix.

Reply:
Yes! Or I can noodle it a bit and submit a suggested fix?